src/utils/time.ts

```diff
@@ -32,10 +48,10 @@ export function subtractTimedelta(date: Date, components: DateTimeComponents) {
 
 export function negateDateTimeComponents(components: DateTimeComponents) {
     const negComponents: DateTimeComponents = {}
-    let key: keyof DateTimeComponents
+    let key: keyof ConcreteDateTimeComponents
```

Comment:
This assertion solves [noUncheckedIndexedAccess](https://www.typescriptlang.org/tsconfig#noUncheckedIndexedAccess).
We assert that the key **can only be one that actually exists** in `DateTimeComponents`. 
However, one can still pass in objects that don't have the type `DateTimeComponents` without Typescript complaining (see time.test.ts), so the assertion is a lie.
@Zsar and me discussed, that if we do the assertion: `let key: keyof ConcreteDateTimeComponents` here, it should actually be true.
Therefore an internal mirror version of `DateTimeComponents` is created, which contains all the legal properties. An object of that type is then checked against at runtime.

The other option we found would be to not assert here and just turn of the corresponding compiler check.

What's your opinion on this @mreiche?

Reply:
I dont get the change, sorry. It is not possible to iterate over object keys based on a type or interface?

Reply:
No, it isn't. TS Types and interfaces only exist at compile time, so we can't iterate over them.
The Iterator DateTimeComponentsIterable is the only thing that exists at runtime, so we iterate over that.
I have, however, found a way to get rid of the iterator by using TS enums, which do exist at runtime and can be iterated over. See latest commit.